packages/i18n-utils/src/locale-context.tsx

```diff
@@ -19,7 +19,7 @@ export const LocaleProvider: React.FC< Props > = ( { children, localeSlug } ) =>
  * Get the current locale slug from the @wordpress/i18n locale data
  */
 function getWpI18nLocaleSlug(): string | undefined {
-	return i18n.getLocaleData()?.[ '' ]?.language;
+	return i18n.getLocaleData()?.[ '' ]?.language || i18n.getLocaleData()?.[ '' ]?.lang;
```

Comment:
I like how this makes it work for Simple and Atomic. Can we add something to the function comment about how this function only works on wpcom sites? The fact it gets the data using `@wordpress/i18n` might lead readers to believe this'll work on core sites too.

Reply:
ohh I did notice that `lang` and probably `language` may contain values like `pt_br` and `zh_tw` which aren't legitimate _locales_. we had the same problem in the ETK php side https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/common/index.php#L164

I'm going to include the provider and pass the locale from php in this case for now, but I think that this should be addressed, I'm not sure in what circumstances this `getWpI18nLocaleSlug` is currently relied on but it look like it would currently return an invalid locale for chinese and portugese users.

Reply:
> `lang` and probably `language` may contain values like pt_br and zh_tw which aren't legitimate locales.

😮 I just double checked this in production and I was wrong, `.language` _doesn't_ appear to be a locale slug!

`.language` can be "zh_TW" and and "pt_BR". When what we really want it "zh-tw" and "pt-br". Am I right about that?

In that case, the property we actually want is `.localeSlug`. Once again, it's still a wpcom only thing, so I'm guessing isn't available on Atomic.


```suggestion
	return i18n.getLocaleData()?.[ '' ]?.localeSlug;
```

`.localeSlug` is _actually_ what Calypso uses in the `i18n-calypso` package: https://github.com/Automattic/wp-calypso/blob/5e0a07e0ddaff970249d36123ff90e940bbacee0/packages/i18n-calypso/src/i18n.js#L273

Reply:
I'd recommend removing the `.lang` from this PR. Maybe we should open another PR to change this to `.localeSlug` too, but we might wanna find a few places where we should test first. This value is often being passed as an API param, especially in Gutenboarding, so that would be a good thing to test when we switch.

Reply:
WordPress.com uses different locale format from core. 

I like to believe the name difference between `lang` and `language` was intentional to prevent mixing these inadvertently.

Because I'd mostly forgotten this stuff I figured I should look it up again and share some links:

WordPress.org uses POSIX locales ([ISO 15897](https://en.wikipedia.org/wiki/ISO/IEC_15897)), most likely because that's what [gettext uses](https://www.gnu.org/software/gettext/manual/html_node/Locale-Names.html) (the underlying thing that loads translations for WordPress)

WordPress.com sticks to what essentially amounts to [BCP-47](https://www.w3.org/International/questions/qa-choosing-language-tags) ([rfc](https://www.rfc-editor.org/rfc/bcp/bcp47.txt)) but in reality it is probably more of a side effect of deferring locale definitions to the [Unicode CLDR project](http://cldr.unicode.org/).

I'm not sure why the divergence though, I think the locale coverage is likely better in CLDR or at least more frequently updated. I think I recall looking into this once and finding gettext defers to debian who then ends up deferring to a transformed subset of CLDR anyway (or something like that - don't quote me.)

Reply:
^ This.

I've been looking for the solution, and I finally noticed '@automattic/languages', which we can use to map the .org locale code to the .com locale slug.

We can reverse what we do to get the .org slug in `../blob/trunk/client/lib/wporg/index.js` [here](https://github.com/Automattic/wp-calypso/blob/63cae7e62ce188e57bb6237309ca4a2b7febe095/client/lib/wporg/index.js#L29)
 
Aside: I quite like the logic we've got in `client/landing/gutenboarding/components/locale-context/index.tsx::getLocale()`, where we check the loaded translations and then fall back to the user's slug. We still seem to have the right value there on atomic and I suspect jetpack will be fine too. That file has several calypso dependencies, though, so I don't think extracting it is a good solution here.